flame/metrics.cpp

```diff
@@ -165,11 +165,15 @@ void MetricsMgr::aggregate(bool no_avgs)
         }
 
         // TODO avg of averages, here be dragons
+        auto rcv_cnt = 0;
         for (const auto &i : _metrics) {
-            _agg_period_response_avg_ms += i->_period_response_avg_ms;
+            if (i->_period_r_count) {
```

Comment:
So we're only including periods when we received responses?

This is meant to fix metrics during the startup phase I guess?

But what will the impact be if there are no responses during a normal run? It seems like this will drop periods and give artificially low response times?  Or is the expectation that a period is small enough that once we start we will never see periods without responses?

Reply:
Right so a period is about 1 second long. This loop calculates the average response ms for that 1 second period across concurrent traffic generators. Prior to this PR, if any of the generators did not receive any responses (but others did), it was including 0s in the average, artificially lowering it. This PR essentially changes the "period response average" metric to mean "the average response time across all traffic generators _that received a response_", which is I think much more in line with user expectations. Without this, it's possible to see the very unexpected result of the average response ms being lower than the minimum.

Reply:
And the fact that no responses were received will (likely) be reflected in the separate timeout metric.